plugins/file/index.js

```diff
@@ -80,13 +83,17 @@ module.exports = {
 
         const type = mimos.type(contentType);
 
-        const extension = type.extensions[0] || "";
-        const filename = `${uuid.v4()}.${extension}`;
-
-        let fileKey = `${getDateString()}/${filename}`;
+        let fileKey;
+        if (fullPath === undefined) {
+          const extension = type.extensions[0] || "";
+          const filename = `${uuid.v4()}.${extension}`;
+          fileKey = `${getDateString()}/${filename}`;
 
-        if (options.keyPrefix) {
-          fileKey = `${options.keyPrefix}${fileKey}`;
+          if (options.keyPrefix) {
+            fileKey = `${options.keyPrefix}${fileKey}`;
+          }
+        } else {
+          fileKey = fullPath;
         }
 
         const params = {
```

Comment:
Could you additionally create a PR to add the missing content types to the q-server-nzz config?- See code here -> https://github.com/nzzdev/Q-server-nzz/blob/b2bf4a91da4ed00770b9f4a78873408884291815/index.js#L46